src/treelearner/ocl/histogram16.cl

```diff
@@ -437,6 +404,8 @@ R""()
             // thread 8, 9, 10, 11, 12, 13, 14, 15 now process feature 0, 1, 2, 3, 4, 5, 6, 7's gradients for example 8, 9, 10, 11, 12, 13, 14, 15
             #if CONST_HESSIAN == 0
             atomic_local_add_f(gh_hist + addr2, stat2);
+            #else
+            atom_inc((uint*)(gh_hist + addr2));
```

Comment:
@huanzhang12 is this added okay? 
or simply `atomic_local_add_f(gh_hist + addr2, 1.0f)` ?

Reply:
If the content at address (gh_hist + addr2) is an integer, we can use atom_inc(). If it is a floating point number, we must use atomic_local_add_f(gh_hist + addr2, 1.0f). I am not sure if you want to store a float or int here?

atomic_local_add_f() can be hundreds times slower than atom_inc(), since there is no native floating point atomics support on GPUs. So using atom_inc() if possible (but we must make sure the content is an integer).

Reply:
could we store it as int firstly and use inc(), then multiply it with hessian[0] and change it to float in-place?